### PR TITLE
fix(chore): resolving sonar smells to improve quality gate

### LINKED ISSUE #142: Sonar Quality Gate Fix
**Describe the bug**
Fix the sonar code quality issues.

**Screenshots**
Refer
https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=sourcefuse_loopback4-authentication
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&id=sourcefuse_loopback4-authentication

### PR DESCRIPTION
**Pull request by arpit1503khanna**
resolving sonar smells to improve quality gate

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #142 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/strategies/SAML/saml-verify.provider.ts [hunk 1]
```diff
@@ -11,10 +11,6 @@ import {VerifyFunction} from '../../types';
  * It will just throw an error saying Not Implemented
  */
 export class SamlVerifyProvider implements Provider<VerifyFunction.SamlFn> {
-  constructor() {
```
Comment (yeshamavani) [inline src/strategies/SAML/saml-verify.provider.ts:14]: ??

File: src/strategies/passport/passport-apple-oauth2/apple-auth-verify.provider.ts [hunk 2]
```diff
@@ -9,8 +9,6 @@ import {VerifyFunction} from '../../types';
 export class AppleAuthVerifyProvider
   implements Provider<VerifyFunction.AppleAuthFn>
 {
-  constructor() {}
```
Comment (yeshamavani) [inline src/strategies/passport/passport-apple-oauth2/apple-auth-verify.provider.ts:12]: why are constructors removed 
how will class be initialised without constructors

File: src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts [hunk 3]
```diff
@@ -29,79 +27,84 @@ export class LocalPasswordStrategyFactoryProvider
...
+        if (!user) {
+          throw new HttpErrors.Unauthorized(AuthErrorKeys.InvalidCredentials);
+        }
+        cb(null, user);
+      } catch (err) {
+        cb(err);
+      }
+    };
+  }
+  getLocalStrategyVerifierUndefinedRequest(
```
Comment (yeshamavani) [inline src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts:65]: name

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: src/strategies/passport/passport-apple-oauth2/apple-auth-verify.provider.ts (hunk 2 above)
Comment (arpit1503khanna) [reply src/strategies/passport/passport-apple-oauth2/apple-auth-verify.provider.ts:12]: It was there in sonar isssue, to remove empty constructors.

File: src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts (hunk 3 above)
Comment (arpit1503khanna) [reply src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts:65]: would 'getDefaultLocalStrategy' be work?

**Comment by yeshamavani** [comment]
sonar code smells

**Review by yeshamavani — APPROVED** [review]

File: src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts (hunk 3 above)
Comment (yeshamavani) [reply src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts]: yes

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**